pkg/deploy/envParams.go

```diff
@@ -50,6 +64,9 @@ func ApplyParams(componentPath string, imageParamsMap map[string]string, isUpdat
 	for i := range paramsEnvMap {
 		relatedImageValue := os.Getenv(imageParamsMap[i])
 		if relatedImageValue != "" {
+			if paramsEnvMap[i] != relatedImageValue {
+				fmt.Printf("%s:%s: replacing %s with %s\n", paramsFile, i, paramsEnvMap[i], relatedImageValue)
+			}
```

Comment:
any reason to add this one?
it will be printed out during any Operator pod restart/clean installation, right?

Reply:
> any reason to add this one? it will be printed out during any Operator pod restart/clean installation, right?

Mostly to debug such cases like the original report, info only. Yes, it will be printed for the actual replacement which should happen only once. Basically, we can avoid file manipulations if there is no replacements. WDYT?

Reply:
yeah, thats what i was thinking this morning, 
so i draft a https://github.com/opendatahub-io/opendatahub-operator/pull/1181
the main idea is to :  "check" -> if missing->"call ApplyParams"
but for the workbench cases: since it does happen (even rarely) so it is done with a "post-check" https://github.com/opendatahub-io/opendatahub-operator/pull/1177

Reply:
Ok, I can then add it on top on this since it has the condition check already.
For #1177 I'm not convinced that it is the case. The PR is pretty hackish.